### PR TITLE
[Issue #126] base implementation of storages based on django-storages

### PR DESCRIPTION
**Pull request by mattiagiupponi**
Base implementation of cloud storages for dropbox, gcs and aws

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [ ] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [ ] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [ ] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Comment by afabiani** [comment]
@mattiagiupponi nice, two things:

1. Put the headers on new files (see the `__init.py` as an instance)
2. Spend some time to add some tests

**Comment by giohappy** [comment]
@mattiagiupponi @afabiani why are you pushing towards our own repo?
And just for curiosity... will be there a local sotrage implementation? :D

**Comment by mattiagiupponi** [comment]
> @mattiagiupponi @afabiani why are you pushing towards our own repo?

We are using it as a test branch to start putting some code

> And just for curiosity... will be there a local sotrage implementation? :D

Yes, is available here https://github.com/geosolutions-it/geonode/blob/162c1658726273ade2c495ee120b69a396e78b79/geonode/storage/manager.py#L91

**Comment by mattiagiupponi** [comment]
> @mattiagiupponi nice, two things:
> 
> 1. Put the headers on new files (see the `__init.py` as an instance)
> 2. Spend some time to add some tests

I have added some tests to verify that the wrappers are correctly called the methods of the library `django-storages`. is this enough @afabiani ?

File: requirements.txt [hunk 1]
```diff
@@ -107,6 +107,9 @@ django-bootstrap3-datetimepicker-2==2.8.2
 
 # AWS S3 dependencies
 django-storages==1.11.1
+dropbox==11.9.0
```
Comment (afabiani) [inline requirements.txt:110]: @mattiagiupponi make sure we put the dependencies on "setup.cfg" file too

**Review by afabiani — APPROVED** [review]

**Pull request merged**